integr8ly/cloud-resource-operator
# add phase status metric to aws resources

## Overview

Jira: https://issues.redhat.com/browse/INTLY-6521

## Verification
BYOC required
- Run ```make cluster/prepare```
- Run ```make cluster/seed/managed/postgres cluster/seed/managed/redis cluster/seed/managed/blobstorage```
- Run ```make run/local```
- Wait for the redis and postgres to reconcile 
- create snapshots from the following cr for redis and posgres 
```yaml
apiVersion: integreatly.org/v1alpha1
kind: RedisSnapshot
metadata:
  name: example
  namespace: cloud-resource-operator
spec: 
  resourceName: example-redis

```
```yaml
apiVersion: integreatly.org/v1alpha1
kind: PostgresSnapshot
metadata:
  name: example
  namespace: cloud-resource-operator
spec: 
  resourceName: example-postgres
```
- Run ```curl localhost:8383/metrics | grep aws```
- Ensure the following metrics are there 
  - redhat_rhmi_postgres_status_phase 
  - redhat_rhmi_postgres_snapshot_status_phase
  - redhat_rhmi_blobstorage_status_phase
  - redhat_rhmi_redis_status_phase
  - redhat_rhmi_redis_snapshot_status_phase
